infrastructure/snyk_api/snyk_api.go

```diff
@@ -78,28 +91,38 @@ func NewSnykApiClient(client func() *http.Client) SnykApiClient {
 
 func (s *SnykApiClientImpl) SastSettings() (SastResponse, error) {
 	method := "SastSettings"
+	var response SastResponse
 	log.Debug().Str("method", method).Msg("API: Getting SastEnabled")
 	path := "/cli-config/settings/sast"
 	organization := config.CurrentConfig().Organization()
 	if organization != "" {
 		path += "?org=" + url.QueryEscape(organization)
 	}
-	responseBody, err := s.doCall("GET", path, nil)
+
+	err := s.processApiResponse(method, path, &response)
 	if err != nil {
-		fmtErr := fmt.Errorf("%v: %v", err, responseBody)
-		log.Err(fmtErr).Str("method", method).Msg("error when calling sastEnabled endpoint")
+		log.Err(err).Str("method", method).Msg("error when calling sastEnabled endpoint")
 		return SastResponse{}, err
 	}
+	return response, err
+}
 
-	var response SastResponse
-	unmarshalErr := json.Unmarshal(responseBody, &response)
-	if unmarshalErr != nil {
-		fmtErr := fmt.Errorf("%v: %v", err, responseBody)
-		log.Err(fmtErr).Str("method", method).Msg("couldn't unmarshal SastResponse")
-		return SastResponse{}, err
+func (s *SnykApiClientImpl) FeatureFlagSettings(featureFlagType FeatureFlagType) (FFResponse, error) {
```

Comment:
The argument should just be a string, since we don't want to pre-define all supported feature flags here. Or at least I don't think we do, I don't see a use for that. Let me know if I'm missing something!

Reply:
Form `/check` there is an ENUM defining the FF names so I decided to define a list for them here too for robustness.

I decided to go with the `FeatureFlagType` because we have a predefined `enum` that centralises the names of our feature flags in the repo defining the FFs 🙌